doc/scheduled-configurations/scheduled-configurations-hld.md

```diff
@@ -0,0 +1,547 @@
+
+For example, an administrator may create a time range to run every weekday from 8 AM to 5 PM. Then the administrator will create a configuration to apply an ACL in order to block (or allow) specific IP addresses and bind it to the time range. This ACL will be applied at the beginning of the time range (8 AM) and removed at the end of the time range (5 PM).
+
+The Scheduled Configurations feature enhances the existing SONiC architecture by introducing two new managers within the swss container, alongside the established `cron` utility. This approach not only ensures seamless integration with the current system but also maintains flexibility for future adaptations, as `cron` can be substituted with other scheduling utilities if more advanced or different functionality is required.
+
+![Architecture Design](images/scheduled-configuration-architecture-no-connections.png)
+
+### New Components
+
+#### REDIS DB
+
+##### CONFIG_DB
+
+- **TIME_RANGE**: A table representing the time ranges that were configured by the user.
+- **SCHEDULED_CONFIGURATIONS**: Represents a configuration that is will be applied when the associated time range is enabled. Contains a field that binds this configuration to a time range.
+
+##### STATE_DB
+
+- **TIME_RANGE_STATUS_TABLE**: Table representing whether this time range is currently active.
+
+#### SWSS
+
+- **timerangemgrd**: This application will monitor and manage `TIME_RANGE` table in the `CONFIG_DB`. It will parse, validate, and create a new entry under the `TIME_RANGE_STATUS_TABLE` in the `STATE_DB`, as well as create crontab files that manage the statuses of `TIME_RANGE_STATUS_TABLE`.
+- **cron**: This application will manage the `TIME_RANGE_STATUS_TABLE` for each entry, and update the status based on the scheduled time.
+- **scheduledconfigmgrd**: This application subscribes to `TIME_RANGE_STATUS_TABLE` in the `STATE_DB`, updated by the `cron` application. When the time range is becomed *active*, It will be responsible for applying configurations given by `SCHEDULED_CONFIGURATIONS` in `CONFIG_DB`, by publishing them to the `APPL_DB`, and also removing them when the time range becomes *inactive*.
+
+### Subscription APPL_DB
+
+In order to fully integrate with the new scheduled configurations feature, all switch components desiring to utilize this functionality must subscribe to the `APPL_DB`. The table schema will be the same as the schema used in `CONFIG_DB` to apply new configurations, all that is needed is to add the new subscription to the orchagent subcomponents. This is done in order to distinguish between administrator configurations, and configurations applied due to an application (ie. *Scheduled Configurations*).
```

Comment:
not sure if this assumption is true that "all switch components desiring to utilize this functionality must subscribe to the `APPL_DB`". @prsunny and @dgsudharsan

Reply:
as the scheduled configuration itself is configuration, applying the embedded configuration should not come from the config_db , if scheduled configuration need to be change it will be done in config_db to the scheduled configuration.
if this assumption is accepted than the embedded configuration must not pass using config db, a natural candidate is app_db as the scheduled configuration itself is a sort of application ( focused on setting configuration is a certain time)

Reply:
I am not sure what the issue is with using the `APPL_DB`. If another management application outside of orchagent needs access to the table in the `APPL_DB` then it can subscribe to is since all *mgr applications inherit from "orch".

There also should not be any collisions when configuring through the `APPL_DB` since there is no table schema from `CONFIG_DB` used in `APPL_DB`.